test/extensions/common/wasm/context_test.cc

```diff
@@ -0,0 +1,251 @@
+  EXPECT_EQ(ctx_.getRequestStreamInfo(), &decoder_si);
+  ctx_.setDecoderFilterCallbacksPtr(nullptr);
+
+  EXPECT_EQ(ctx_.getConstRequestStreamInfo(), &access_log_si);
+  ctx_.setAccessLogStreamInfoPtr(nullptr);
+
+  EXPECT_EQ(ctx_.getConstRequestStreamInfo(), &read_filter_si);
+  EXPECT_EQ(ctx_.getRequestStreamInfo(), &read_filter_si);
+  ctx_.setNetworkReadFilterCallbacksPtr(nullptr);
+
+  EXPECT_EQ(ctx_.getConstRequestStreamInfo(), &write_filter_si);
+  EXPECT_EQ(ctx_.getRequestStreamInfo(), &write_filter_si);
+  ctx_.setNetworkWriteFilterCallbacksPtr(nullptr);
+}
+
+TEST_F(ContextTest, GetConnectionTest) {
+  Http::MockStreamEncoderFilterCallbacks encoder_callbacks;
+  Envoy::Network::MockConnection encoder_connection;
+  EXPECT_CALL(encoder_callbacks, connection())
+      .Times(1)
+      .WillRepeatedly(testing::Return(
+          makeOptRef(dynamic_cast<const Network::Connection&>(encoder_connection))));
+  ctx_.setEncoderFilterCallbacksPtr(&encoder_callbacks);
+
+  Http::MockStreamDecoderFilterCallbacks decoder_callbacks;
+  Envoy::Network::MockConnection decoder_connection;
+  EXPECT_CALL(decoder_callbacks, connection())
+      .Times(1)
+      .WillRepeatedly(testing::Return(
+          makeOptRef(dynamic_cast<const Network::Connection&>(decoder_connection))));
+  ctx_.setDecoderFilterCallbacksPtr(&decoder_callbacks);
+
+  Envoy::StreamInfo::MockStreamInfo access_log_si;
+  ctx_.setAccessLogStreamInfoPtr(&access_log_si);
+
+  Network::MockReadFilterCallbacks read_filter_callbacks;
+  Envoy::Network::MockConnection read_filter_connection;
+  EXPECT_CALL(read_filter_callbacks, connection())
+      .Times(1)
+      .WillRepeatedly(testing::ReturnRef(read_filter_connection));
+  ctx_.setNetworkReadFilterCallbacksPtr(&read_filter_callbacks);
+
+  Network::MockWriteFilterCallbacks write_filter_callbacks;
+  Envoy::Network::MockConnection write_filter_connection;
+  EXPECT_CALL(write_filter_callbacks, connection())
+      .Times(1)
+      .WillRepeatedly(testing::ReturnRef(write_filter_connection));
+  ctx_.setNetworkWriteFilterCallbacksPtr(&write_filter_callbacks);
+
+  EXPECT_EQ(ctx_.getConnection(), &encoder_connection);
+  ctx_.setEncoderFilterCallbacksPtr(nullptr);
+
+  EXPECT_EQ(ctx_.getConnection(), &decoder_connection);
+  ctx_.setDecoderFilterCallbacksPtr(nullptr);
+
+  EXPECT_EQ(ctx_.getConnection(), &read_filter_connection);
+  ctx_.setNetworkReadFilterCallbacksPtr(nullptr);
+
+  EXPECT_EQ(ctx_.getConnection(), &write_filter_connection);
+  ctx_.setNetworkWriteFilterCallbacksPtr(nullptr);
+}
+
+TEST_F(ContextTest, SerializeValueMapTest) {
+  std::string result;
+  CelValue value = CelValue::CreateMap(&mock_cel_map_);
+  EXPECT_CALL(mock_cel_map_, ListKeys())
+      .WillOnce(testing::Return(absl::UnimplementedError("CelMap::ListKeys is not implemented")));
+  EXPECT_EQ(serializeValue(value, &result), WasmResult::SerializationFailure);
+
+  EXPECT_CALL(mock_cel_list_, MockIndexOperator(_))
```

Comment:
Can these tests use direct values, or is mocking the only way? Some of the cel-cpp tests use fakes ([example](https://github.com/google/cel-cpp/blob/42c7d1b31cebc36d4f06525d1c1101735a0f6a22/eval/public/builtin_func_test.cc#L1065)), which might be more readable.

If this ends up being more code or extra toil, feel free to ignore.

Reply:
I think that mocking seems to be easier way to test edge cases without growing codes.